### PR TITLE
feat(nuxi): support `--dotenv` for `dev`, `build` and `preview` commands

### PR DESCRIPTION
**Pull request by cpreston321**
### 🔗 Linked issue

Issue: #7646 

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [X] 👌 Enhancement (improving an existing functionality like performance)
- [X] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Wanted to help with stubbing out and adding `--dotenv` to `dev`, `preview` and `build` but for build I would think this would be an override during build but I could not find and overrides so I left `build` out for now until I can learn the best way to override it.

- `nuxi dev`
  - [**Added]** `[--dotenv]` option - relative to rootDir defaults to `rootDir`
- `nuxi preview`
  - [**Added]** `[--dotenv]` option - relative to rootDir defaults to `rootDir`
- `nuxi build`
  - [**Added]** `[--dotenv]` option - relative to rootDir defaults to `rootDir`
  
I will update docs once I have more information! 

Thanks 💚 

### 📝 Checklist

- [X] I have linked an issue or discussion.
- [X] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by cpreston321** [comment]
@pi0 Thanks for the response! I added it to `nuxi build` and updated docs accordingly

File: packages/nuxi/src/commands/dev.ts [hunk 1]
```diff
@@ -40,7 +40,8 @@ export default defineNuxtCommand({
     const rootDir = resolve(args._[0] || '.')
     showVersions(rootDir)
 
-    await setupDotenv({ cwd: rootDir })
+    const dotenvDir = args.dotenv ? resolve(rootDir, args.dotenv) : rootDir
+    await setupDotenv({ cwd: dotenvDir })
```
[the remark this answers is not in the document]
Comment (cpreston321) [reply packages/nuxi/src/commands/dev.ts:44]: @pi0 So do we want to treat `fileName` as  a relative path + name to `rootDir` or do want to add more checks to `--dotenv` to make sure that file exist if not fallback to `.env` for filename ?

[the remark this answers is not in the document]
Comment (pi0) [reply packages/nuxi/src/commands/dev.ts:44]: Actually `setupDotenv` already makes it relative to rootdir. We can still use `--dotenv ./config/.env.staging` with the proposed change.

[the remark this answers is not in the document]
Comment (pi0) [reply packages/nuxi/src/commands/dev.ts:44]: (option name is little confusing. `fileName` can also be a path)

[the remark this answers is not in the document]
Comment (cpreston321) [reply packages/nuxi/src/commands/dev.ts]: @pi0 should be good to go now based on your suggestion.

**Review by pi0 — APPROVED** [review]
Thanks <3

**Pull request merged**